configuration/external_parameters.rst

```diff
@@ -80,24 +123,51 @@ You can now reference these parameters wherever you need them.
 
         <doctrine:config>
             <doctrine:dbal
-                driver="pdo_mysql"
-                dbname="symfony_project"
-                user="%database.user%"
-                password="%database.password%"
+                host="%env(DATABASE_HOST)%"
             />
         </doctrine:config>
 
     .. code-block:: php
 
         $container->loadFromExtension('doctrine', array(
             'dbal' => array(
-                'driver'   => 'pdo_mysql',
-                'dbname'   => 'symfony_project',
-                'user'     => '%database.user%',
-                'password' => '%database.password%',
+                'host' => '%env(DATABASE_HOST)%',
```

Comment:
I propose the following:

 * Keep the "Environment variables" section title and document the `%env(...)%` syntax in there. Explain what by reference means ("You can reference environment variables using the `%env(...)%` syntax. The actual value will be resolved when using the container: [...]").
 * Add a small `.. note::` directive documenting `SYMFONY__` (I personally don't see any reason to use this, other than BC. In other words, why not deprecate it?). This will say something like: "You can also directly set parameters by prefixing the environment variable with `SYFMONY__` (e.g. `SYMFONY__DATABASE__USER` would set a `database.user` parameter). This value will be resolved when warming up the cache and will not change at runtime."